components/micro-gateway-core/src/main/ballerina/src/gateway/utils/analytics_util.bal

```diff
@@ -1,141 +1,159 @@
+    boolean isSecured = <boolean>context.attributes[IS_SECURED];
+    ThrottleAnalyticsEventDTO eventDto = {};
+    var api_Version = "";
 
-// function populateFaultAnalyticsDTO(http:FilterContext context, error err) returns (FaultDTO) {
-//     boolean isSecured = <boolean>context.attributes[IS_SECURED];
-//     FaultDTO eventDto = {};
-//     time:Time time = time:currentTime();
-//     int currentTimeMills = time.time;
-//     json metaInfo = {};
-//     eventDto.apiContext = getContext(context);
-//     var api_Version = apiConfigAnnotationMap[getServiceName(context.serviceName)].apiVersion;
-//     eventDto.apiVersion = (api_Version is string) ? api_Version : "";
-//     eventDto.apiName = getApiName(context);
-//     var resource_Path = getResourceConfigAnnotation(resourceAnnotationMap[context.resourceName] ?: []).path;
-//     eventDto.resourcePath = (resource_Path is string) ? resource_Path : "";
-//     eventDto.method = <string>context.attributes[API_METHOD_PROPERTY];
-//     eventDto.errorCode = <int>runtime:getInvocationContext().attributes[ERROR_RESPONSE_CODE];
-//     eventDto.errorMessage = err.reason();
-//     eventDto.faultTime = currentTimeMills;
-//     eventDto.apiCreatorTenantDomain = getTenantDomain(context);
-//     eventDto.hostName = retrieveHostname(DATACENTER_ID, <string>context.attributes[HOSTNAME_PROPERTY]);
-//     eventDto.protocol = <string>context.attributes[PROTOCOL_PROPERTY];
-//     if (isSecured && context.attributes.hasKey(AUTHENTICATION_CONTEXT)) {
-//         AuthenticationContext authContext = <AuthenticationContext>context.attributes[AUTHENTICATION_CONTEXT];
-//         metaInfo.keyType = authContext.keyType;
-//         eventDto.consumerKey = authContext.consumerKey;
-//         eventDto.apiCreator = authContext.apiPublisher;
-//         eventDto.userName = authContext.username;
-//         eventDto.applicationName = authContext.applicationName;
-//         eventDto.applicationId = authContext.applicationId;
-//         eventDto.userTenantDomain = authContext.subscriberTenantDomain;
-//     } else {
-//         metaInfo.keyType = PRODUCTION_KEY_TYPE;
-//         eventDto.consumerKey = ANONYMOUS_CONSUMER_KEY;
-//         var api_Creater = apiConfigAnnotationMap[getServiceName(context.serviceName)].publisher;
-//         eventDto.apiCreator = (api_Creater is string) ? api_Creater : "";
-//         eventDto.userName = END_USER_ANONYMOUS;
-//         eventDto.applicationName = ANONYMOUS_APP_NAME;
-//         eventDto.applicationId = ANONYMOUS_APP_ID;
-//         eventDto.userTenantDomain = ANONYMOUS_USER_TENANT_DOMAIN;
-//     }
-//     metaInfo.correlationID = <string>context.attributes[MESSAGE_ID];
-//     eventDto.metaClientType = metaInfo.toString();
-//     return eventDto;
-// }
+    APIConfiguration? apiConfiguration = apiConfigAnnotationMap[getServiceName(context.getServiceName())];
+    if (apiConfiguration is APIConfiguration) {
+      api_Version = apiConfiguration.apiVersion;
```

Comment:
This variable is not used. This variable contains the actual value for the version and it should be made equal to the apiVersion variable